### PR TITLE
bugfix-7801 Fixing validation bugs in experiments, models modals

### LINKED ISSUE #7801: [BUG] Create experiment modal validations are not working as expected
### Willingness to contribute

Yes. I can contribute a fix for this bug independently.

### MLflow version

2.1.1

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Darwin Kernel Version 22.2.0
- **Python version**: Python 3.9.16
- **yarn version, if running the dev UI**:

### Describe the problem

Create Experiment modal validations are not working as expected. Modal allows users to click on `create` button without entering experiment name. UI gets stuck and cannot close the dialog/modal.

### Steps to reproduce the bug

Open Create Experiment screen.
Click create without entering experiment name.

https://user-images.githubusercontent.com/1637727/217169375-cce91873-e294-4806-91ef-e6819b3fa7ed.mov

### Code to generate data required to reproduce the bug

_No response_

### Is the console panel in DevTools showing errors relevant to the bug?

```
index.js:11 TypeError: Cannot read properties of undefined (reading 'length')
    at validations.js:6:15
    at y (debounce.js:95:19)
    at x (debounce.js:144:14)
    at w (debounce.js:132:14) "Uncaught TypeError: Cannot read properties of undefined (reading 'length')"
window.onerror @ index.js:11
error (async)
(anonymous) @ index.js:16
(anonymous) @ index.js:13
(anonymous) @ index.js:13
validations.js:6 Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at validations.js:6:15
    at y (debounce.js:95:19)
    at x (debounce.js:144:14)
    at w (debounce.js:132:14)
```

### Does the network panel in DevTools contain failed requests relevant to the bug?

_No response_

### PR DESCRIPTION
**Pull request by subramaniam02**
Signed-off-by: subraman <subramaniam02@gmail.com>

## Related Issues/PRs

 #7801
closes #7801

## What changes are proposed in this pull request?

Fixed validations and enhanced error msgs 
- Empty experiment name
- Empty model name
- Existing experiment name
- Existing model name
- Existing Tag name for model

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

Enhanced validation message in create experiment and models modals.

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [x] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by subramaniam02** [comment]
Screenshots
Screenshot 2023-02-13 at 7 26 18 PM
Screenshot 2023-02-13 at 7 26 29 PM
Screenshot 2023-02-13 at 7 26 53 PM
Screenshot 2023-02-13 at 7 27 03 PM
Screenshot 2023-02-13 at 7 27 10 PM

File: mlflow/server/js/src/common/components/EditableTagsTableView.js [hunk 1]
```diff
@@ -92,6 +92,8 @@ export class EditableTagsTableViewImpl extends Component {
                     description:
                       'Error message for name requirement in editable tags table view in MLflow',
                   }),
+                },
+                {
```
Comment (dbczumar) [inline mlflow/server/js/src/common/components/EditableTagsTableView.js:96]: Is this change necessary?

**Review by dbczumar — APPROVED** [review]
LGTM with a small question. Thanks @subramaniam02 !

File: mlflow/server/js/src/common/components/EditableTagsTableView.js (hunk 1 above)
Comment (subramaniam02) [reply mlflow/server/js/src/common/components/EditableTagsTableView.js:96]: yes, error msg is not appropriate when tag name already exists.

Screenshot 2023-02-14 at 4 18 19 PM

Comment (dbczumar) [reply mlflow/server/js/src/common/components/EditableTagsTableView.js:96]: Got it - thank you!

**Pull request merged**